internal/policy/root.go

```diff
@@ -27,35 +29,49 @@ func InitializeRootMetadata(key *tuf.Key) *tuf.RootMetadata {
 	return rootMetadata
 }
 
-// AddTargetsKey adds targetsKey as a trusted public key in rootMetadata for the
+// AddTargetsKey function is used to add the 'targetsKey' as a trusted public key in 'rootMetadata' for the
 // top level Targets role.
-func AddTargetsKey(rootMetadata *tuf.RootMetadata, targetsKey *tuf.Key) *tuf.RootMetadata {
+func AddTargetsKey(rootMetadata *tuf.RootMetadata, targetsKey *tuf.Key) (*tuf.RootMetadata, error) {
+	if rootMetadata == nil {
+		return nil, fmt.Errorf("rootMetadata is nil")
+	}
+	if targetsKey == nil {
+		return nil, fmt.Errorf("targetsKey is nil")
+	}
+
 	rootMetadata.Keys[targetsKey.KeyID] = targetsKey
+
 	if _, ok := rootMetadata.Roles[TargetsRoleName]; !ok {
 		rootMetadata.AddRole(TargetsRoleName, tuf.Role{
 			KeyIDs:    []string{targetsKey.KeyID},
 			Threshold: 1,
 		})
-		return rootMetadata
+		return rootMetadata, nil
 	}
 
 	targetsRole := rootMetadata.Roles[TargetsRoleName]
 	for _, keyID := range targetsRole.KeyIDs {
 		if keyID == targetsKey.KeyID {
-			return rootMetadata
+			return rootMetadata, nil
 		}
 	}
 
 	targetsRole.KeyIDs = append(targetsRole.KeyIDs, targetsKey.KeyID)
 	rootMetadata.Roles[TargetsRoleName] = targetsRole
 
-	return rootMetadata
+	return rootMetadata, nil
 }
 
-// DeleteTargetsKey removes keyID from the list of trusted top level Targets
-// public keys in rootMetadata. It does not remove the key entry itself as it
-// does not check if other roles can be verified using the same key.
+// DeleteTargetsKey removes 'keyID' from trusted public keys for top level Targets
```

Comment:
nit: 80 cols on docstrings

Reply:
bump, this is still over 80 cols per my editor

![image](https://github.com/gittuf/gittuf/assets/8928778/33fb5074-9dd2-4df4-afd4-30b281a65949)